common/src/main/java/com/microsoft/identity/common/internal/authorities/AzureActiveDirectoryAuthority.java

```diff
@@ -173,4 +173,15 @@ public AzureActiveDirectoryAudience getAudience() {
         return mAudience;
     }
 
+    /**
+     * Checks if current authority belongs to the same cloud as the passed in authority
+     * @param authorityToCheck authority to check against
+     * @return true if the preferred network host name matches for both authorities, otherwise false
+     */
+    public boolean isSameCloudAsAuthority(@NonNull final AzureActiveDirectoryAuthority authorityToCheck){
+        final AzureActiveDirectoryCloud currentAuthorityCloud = AzureActiveDirectory.getAzureActiveDirectoryCloud(this.getAuthorityURL());
+        final AzureActiveDirectoryCloud authorityToCheckCloud = AzureActiveDirectory.getAzureActiveDirectoryCloud(authorityToCheck.getAuthorityURL());
+        return currentAuthorityCloud.getPreferredNetworkHostName().equals(authorityToCheckCloud.getPreferredNetworkHostName());
```

Comment:
we should check if we can do the following:

```java
  return this.getAzureActiveDirectoryCloud().equals(authorityToCheck.getAzureActiveDirectoryCloud())
```

this would mean exposing the `getAzureActiveDirectoryCloud` method and also adding `equals` to the class `AzureActiveDirectoryCloud`

Reply:
I agree with Shahzaib's suggestion.

Reply:
If we're overriding the equals method, I'd like to make sure that we're not going to cause any unforeseen behavior later on.  If we implement equals (and almost every class we create should implement equals unless there is a very good reason), we *must* implement hashCode, and the two *must* be consistent.

Reply:
+1...we can probably just use Lombok's annotations in most cases and let it do the work.

Reply:
I updated the code to use getAuthorityURL as that includes the preferred_network_host_name.

Reply:
If you look at getAuthorityURL it returns the preferred_network_host_name, which should take care of alias checking

The thing I am still debating in my head is should we expose getAzureActiveDirectoryCloud from this class or should it be just the preferred_network_host_name which is what is being used for cloud comparison and is already being returned in getAuthorityURL.
Thoughts?

When we want to have equals method in the AzureActiveDirectoryCloud class.. what all should it compare?

Reply:
I would personally compare the following:

```java
    @SerializedName("preferred_network")
    private final String mPreferredNetworkHostName;

    @SerializedName("preferred_cache")
    private final String mPreferredCacheHostName;

    @SerializedName("aliases")
    private List<String> mCloudHostAliases;
```

then you know the clouds are truly equal...the worrying this is that `mCloudHostAliases` is a List...we should refactor that info a `Set`

Reply:
Why is that worrying?  Does it actually matter?

Reply:
I am not sure if comparing aliases in needed to ensure if a authority belongs to same cloud as another authority, as long as the preferred_network matches. 
Could there be cases where preferred_network but not the aliases?

Reply:
@shahzaibj can you think of a scenario where the preferred_network name is same but the alias list is different?  

I am inclined to not expose extra properties if we don’t need to.